src/backend/storage/data_table.cpp

```diff
@@ -326,6 +347,53 @@ bool DataTable::InsertInSecondaryIndexes(const storage::Tuple *tuple,
   return true;
 }
 
+/**
+ * @brief Check if all the foreign key constraints on this table
+ * is satisfied by checking whether the key exist in the referred table
+ *
+ * FIXME: this still does not guarantee correctness under concurrent transaction
```

Comment:
This still does not guarantee correctness under concurrent transaction because I only check if the key exists the referred table's index -- however this key might be a uncommitted key that is not visible to others and it might be deleted if that txn abort 
-- We should modify this function and add logic to check if the result of the ScanKey is visible.